src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangRecordProcessor.java

```diff
@@ -136,6 +141,24 @@ public void shutdown(ShutdownInput shutdownInput) {
         }
     }
 
+    @Override
+    public void shutdownRequested(IRecordProcessorCheckpointer checkpointer) {
+        LOG.info("Shutdown is requested.");
+        if (!initialized) {
+            LOG.info("Record processor was not initialized so no need to initiate a final checkpoint.");
+            return;
+        }
+        try {
+            LOG.info("Requesting a checkpoint on shutdown notification.");
+            checkpointer.checkpoint();
```

Comment:
Whether to checkpoint is determined by the application, so checkpointing here isn't safe.  This should use a similar dispatch mechanism that processRecords uses.  It might be possible to shortcut the dispatch at this time, and just make whether to checkpoint a configuration flag.

The [dispatch code for processRecords](https://github.com/awslabs/amazon-kinesis-client/blob/master/src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangProtocol.java#L84) for reference.